ddtrace/contrib/asgi/middleware.py

```diff
@@ -288,6 +289,9 @@ async def wrapped_blocked_send(message):
             try:
                 core.dispatch("asgi.start_request", ("asgi",))
                 return await self.app(scope, receive, wrapped_send)
+            except BlockingException as e:
```

Comment:
<3

I was just talking to Emmett about using a pattern like this vs dispatch_with_results, I really like it specifically for request blocking stuff.